core/src/main/java/org/apache/carbondata/core/scan/filter/FilterUtil.java

```diff
@@ -2265,7 +2265,8 @@ public static int compareValues(byte[] filterValue, byte[] minMaxBytes,
       defaultValue = FilterUtil
           .getMaskKey(key, currentBlockDimension, segmentProperties.getSortColumnsGenerator());
     } else {
-      defaultValue = ByteUtil.toXorBytes(key);
+      defaultValue = FilterUtil
+          .getMaskKey(key, currentBlockDimension, segmentProperties.getDimensionKeyGenerator());
```

Comment:
`if` and `else` has same code. remove if condition

Reply:
not same code, one is sortColumnsGenerator and other is DimensionKeyGenerator